### PR TITLE
[aws][1password] Fix field conflicts

### LINKED ISSUE #2667: [azure] Mapping conflicts in Azure Fleet Integrations
Hi
Following the integration instructions and guides from Microsoft on Azure Logs v1.0.1
we end up getting multiple conflicts.

Checking out field mappings on the following, shows some inconsistencies.

 .ds-logs-azure.platformlogs = type:ip
 .ds-logs-azure.activitylogs = type:ip
 .ds-logs-azure.auditlogs = type:keyword
 .ds-logs-azure.signinlogs = type:ip

GET /logs-*/_mapping/field/client.ip
},
  ".ds-logs-azure.auditlogs-default-2022.02.09-000001" : {
    "mappings" : {
      "client.ip" : {
        "full_name" : "client.ip",
        "mapping" : {
          "ip" : {
            "type" : "keyword",
            "ignore_above" : 1024

 .ds-logs-azure.platformlogs = type:long
 .ds-logs-azure.activitylogs = type:long
 .ds-logs-azure.auditlogs = type:long
 .ds-logs-azure.eventhub = type:keyword
 .ds-logs-azure.signinlogs = type:long

GET /logs-*/_mapping/field/azure-eventhub.offset
  },
  ".ds-logs-azure.eventhub-default-2022.02.09-000001" : {
    "mappings" : {
      "azure-eventhub.offset" : {
        "full_name" : "azure-eventhub.offset",
        "mapping" : {
          "offset" : {
            "type" : "keyword",
            "ignore_above" : 1024
       
 .ds-logs-azure.platformlogs type=long
 .ds-logs-azure.activitylogs type=long
 .ds-logs-azure.eventhub type=keyword
 .ds-logs-azure.signinlogs type=long
 `
GET /logs-*/_mapping/field/azure-eventhub.sequence_number
},
  ".ds-logs-azure.eventhub-default-2022.02.09-000001" : {
    "mappings" : {
      "azure-eventhub.sequence_number" : {
        "full_name" : "azure-eventhub.sequence_number",
        "mapping" : {
          "sequence_number" : {
            "type" : "keyword",
            "ignore_above" : 1024

Azure Active Directory Audit Logs (eventhub)
 AuditLogs
 SignInLogs
 NonInteractiveUserSignInLogs
 ServicePrincipalSignInLogs
 ManagedIdentitySignInLogs
 ProvisioningLogs
 ADFSSignInLogs
 RiskyUsers
 UserRiskEvents

 Azure Diagnostic Settings / insights-operational-logs

 Administrative
 Security
 ServiceHealth
 Alert
 Recommendation
 Policy
 Autoscale
 ResourceHealth

### LINKED ISSUE #2684: [8.0] AWS & 1Password Integrations are creating Index field conflicts
From a vanilla installation when I enabled the following integrations:
- 1Password
- AWS
- ~Kubernetes~

I get field conflicts when viewed from 'Data View' in Kibana. See screenshots.

CleanShot 2022-02-11 at 16 53 30@2x

CleanShot 2022-02-11 at 16 53 15@2x

CleanShot 2022-02-11 at 16 55 29@2x

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Resolve field conflicts.

NOTE: The first three commits only contain formatting changes and can be verified via [git-generate](https://github.com/rsc/rf/tree/main/git-generate). It's recommended that you ignore those during the review.

### Data Stream Changes

`1password.item_usages`

-  `event.created` - keyword -> date (per ECS)

`1password.signin_attempts`

-  `event.created` - keyword -> date (per ECS)

`aws.elb_logs`

- `source.port` - keyword to long (per ECS)
- `tracing.trace.id` renamed to `trace.id` (per [ECS](https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html#field-trace-id))
- Add missing ECS mappings for several destination, source, event, and http fields.

`azure.auditlogs`

- `client.ip` - keyword to ip (per ECS)

`azure.eventhub`

- `azure-eventhub.offset` - keyword to long
- `azure-eventhub.sequence_number` - keyword to long

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #2684
- Closes #2667

### CODE REVIEW

**Comment by legoguy1000** [comment]
@andrewkroh Should be gtg

File: packages/azure/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.0.2"
+  changes:
+    - description: Fix field mapping conflict
```
Comment (andrewkroh) [inline packages/azure/changelog.yml:3]:
```suggestion
    - description: Fix field mapping conflict in the auditlogs data stream for `client.ip`. Changed `azure-eventhub.offset` and `azure-eventhub.sequence_number` to longs from keyword in the eventhub data stream.
```

**Review by andrewkroh — DISMISSED** [review]
Thanks. I expanded the PR description with a list of changes to help the other reviewers.

**Comment by andrewkroh** [comment]
Looks like the aws.elb_logs needs a `convert` processor for the port values.

> [1] parsing field value failed: field "source.port"'s Go type, string, does not match the expected field type: long (field value: 2817)

**Comment by andrewkroh** [comment]
@elastic/integrations can you please review the AWS and Azure fixes.

**Comment by andrewkroh** [comment]
@kaiyan-sheng I'd consider this a bugfix because the data didn't comply with ECS. Fleet will rollover their data streams given the change to the template and this will prevent mapping conflicts, but there will be a Kibana index pattern conflict for the between the old data and the new data. At least now the new data from these integrations will not conflict with each other in the `logs-*` index pattern.

File: packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -163,6 +163,17 @@ processors:
   - set:
       field: event.end
       value: '{{ @timestamp }}'
+  - convert:
+      field: source.address
+      target_field: source.ip
+      type: ip
+      ignore_missing: true
```
Comment (andrewkroh) [inline packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml:170]: `ignore_missing` is redundant when used with `ignore_failure`. Please remove.
```suggestion
```

File: packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -163,6 +163,17 @@ processors:
...
+      ignore_failure: true
+  - convert:
+      field: source.port
+      type: long
+      ignore_missing: true
```
Comment (andrewkroh) [inline packages/aws/data_stream/elb_logs/elasticsearch/ingest_pipeline/default.yml:175]:
```suggestion
```

**Comment by andrewkroh** [comment]
@Mergifyio update

**Comment by andrewkroh** [comment]
I rebased this so that we can review the fixes separately from the formatting updates to the expected JSON pipeline test files.

**Comment by legoguy1000** [comment]
@andrewkroh thanks. Apologies for not staying on this, been super busy at work.

**Comment by andrewkroh** [comment]
No worries! We realize you have a two jobs working here and there 😉 .

**Review by andrewkroh — APPROVED** [review]
LGTM. Approving on behalf of elastic/security-external-integrations.

@elastic/integrations, can you please add your review for the aws and azure changes.

**Review by kaiyan-sheng — APPROVED** [review]
aws and azure looks good. Thanks!

**Comment by nicpenning** [comment]
How should we handle other field conflicts such as this? New issue?

event.duration is another one for the azure integration:

This is a long everywhere else besides the azure activity logs.

{
  ".ds-logs-azure.activitylogs-default-2022.03.23-000001" : {
    "mappings" : {
      "event.duration" : {
        "full_name" : "event.duration",
        "mapping" : {
          "duration" : {
            "type" : "keyword",
            "ignore_above" : 1024
          }
        }
      }
    }
  },

**Comment by legoguy1000** [comment]
I would do a new issue since this was already merged

**Pull request merged**